src/preprocessor.rs

```diff
@@ -100,7 +85,53 @@ pub fn replace_nonprintable(input: &[u8], tab_width: usize) -> String {
                     output.push(c)
                 }
                 // everything else
-                c => output.push_str(&c.escape_unicode().collect::<String>()),
+                chr => {
+                    if use_caret {
+                        // Replace LF, CR, NUL, BEL, BS, ESC using caret notation
+                        match chr {
+                            // line feed
+                            '\x0A' => {
+                                output.push_str("^J\x0A");
+                                line_idx = 0;
+                            }
+                            // carriage return
+                            '\x0D' => output.push_str("^M"),
+                            // null
+                            '\x00' => output.push_str("^@"),
+                            // bell
+                            '\x07' => output.push_str("^G"),
+                            // backspace
+                            '\x08' => output.push_str("^H"),
+                            // escape
+                            '\x1B' => output.push_str("^["),
+
+                            // everything else
+                            c => output.push_str(&c.escape_unicode().collect::<String>())
```

Comment:
To reduce the diff and levels of indentation I would prefer if you added `if use_caret` to each existing character arm instead is of doing the "refactoring" in the current PR

Reply:
I'm sorry, I don't quite understand your comment. So should I refactor it or not?

Reply:
Sorry for being vague. I mean that I would prefer e.g.

```
'\x1B' => output.push('␛'),
```

to be changed to

```
'\x1B' => output.push_str(if use_caret { "^[" } else  { "␛" }),
```